src/hooks/ProcessedRichNote/ProcessedRichNote.ts

```diff
@@ -22,9 +29,24 @@ export const useProcessedRichNote = (note: string | undefined) => {
    *   [www.wikipedia.org],
    * ]
    */
+  // Add newline before any links (starts with 'https' or 'ipfs') in the note that don't already have them
+  const formattedNote =
+    note
+      ?.split(' ')
+      .map(word => {
+        if (word.match(ipfsLinkRegex)) {
+          return `\n${formatIpfsLink(word)}`
+        } else if (word.match(URLRegex)) {
+          return `\n${word}`
+        }
+        return word
+      })
+      .join(' ') ?? note
+
   const linksPerLine = useMemo(() => {
-    return note?.split('\n').map(line => line.match(URLRegex)) ?? []
-  }, [note])
+    // split from new line OR space
+    return formattedNote?.split('\n').map(line => line.match(URLRegex)) ?? []
```

Comment:
@johnnyd-eth somehow I'm expecting this to break. what happens in the case where:

- user uploads image
- user adds sticker
- user includes HTTP image link and IPFS image link in the text memo

Reply:
You're right. Will look into this